bin/main.rb

```diff
@@ -1,44 +1,110 @@
+      symbol.capitalize!
     end
+    player1 = Player.new(@name_1, 1, symbol)
+    return player1
+  end
 
-    UserInteraction.get_rol
-
-    def self.input_cell
-        @cell = nil
-        puts "\nSelect a number in the cell"
-        @cell = gets.chomp.to_i
+  def setPlayer2(player1)
+    if player1.symbol == "X"
+      player2 = Player.new(@name_2, 2, 'O')
+    else
+      player2 = Player.new(@name_2, 2, 'X')
     end
+    return player2;
+  end
 
-    UserInteraction.input_cell
+  def option(player, board, game)
+    result = false
+    puts board.display_board
+    puts "\n#{player.name} is playing!"
+    puts "Please select one cell between 1 to 9: "
+    move = gets.chomp
+    while !game.options.include? move.to_i
+      print "\nselect a number from 1 to 9: "
+      move = gets.chomp
+    end
+    while !board.add_map(move.to_i, player.symbol.to_s)
+      print "\nInvalid Option, select again: "
+      move = gets.chomp
+    end
+    if player.value == 1
+      board.winner = 1 if board.check_map(move.to_i, player.symbol.to_s)
+    else
+      board.winner = 2 if board.check_map(move.to_i, player.symbol.to_s)
+    end
+    result = true if board.winner != 0
+    result = true if board.plays == 9
+    print board.display_board if player == 1 && result
+    return result
+  end
 
-    def self.again?
-        puts "##################"
-        puts "\n#PLAY AGAIN? (y/n)"
-        puts "##################"
-        @again = gets.chomp
-        if @again === "y" || @again === "Y"
-            validate_rol
-        elsif @again === "n" || @again === "N"
-            puts "Adios Amigo!"
-            exit
-        else
-            puts "Invalid option, please try again! "
-            again?
-        end
+  def endGame(game, board)
+    print board.display_board
+    if board.winner == 1
+      print "\n ***** #{@name_1.capitalize} Wins ***** \n"
+    else board.winner == 2
+      print "\n ***** #{@name_2.capitalize} Wins ***** \n"
+    
+    end
+    
+    while @again == false do
+      puts "\n Do you want to play again? (y/n) :  "
+      ans = gets.chomp
+      if ans == "y" || ans == "Y"
+        return true
+      elsif ans == "n" || ans == "N"
+        puts "\n\n\nADIOS AMIGO!!"
+        puts "*******************************************"
+        puts "Authors: Asha Abdullahi and Julian Belmonte"
+        puts "*******************************************"
+        exit
+      else
+        puts "Invalid option, please try again!"
+        false
+      end
     end
+  end
+end
+
+main = Main.new
+game = Game.new()
+game.play_again = true
 
-    UserInteraction.again?
+while game.play_again
+  game.play_again = false
+  player1 = main.setPlayer1(game)
+  player2 = main.setPlayer2(player1)
+  board = Board.new()
+  while true
+    break if main.option(player1, board, game)
+    break if main.option(player2, board, game)
+  end
+  if main.endGame(game, board)
+    game.play_again = true
+    board.plays = 0
+  end
```

Comment:
This logic should not be here, this is the game loop can we put it among the game logic? you could create a game start or game loop method and just call it here